### PR TITLE
Introduce toggles and new consent wording to the Emails & Marketing page

### PR DESCRIPTION
**Pull request by vlbee**
## What does this change?

Replaces #784 

Introduce toggles to the Emails & Marketing page - Supporter Exclusive section and below. 
[Figma designs (first iteration)](https://www.figma.com/file/WOztYf2GIbvowlvJeB3vmH/Identity---MMA?node-id=207%3A5892)

IDAPI Api Response model has been updated to reflect that a Consent description is not always returned. 

[Until changes to the API ]((https://github.com/guardian/identity/pull/2079))are released, we are hardcoding the new consent wordings. This will fixed to use the API response as soon as released. 

## How to test

New unit tests added for MarketingToggle and OptOutSection

- [x] Tested in CODE with Idapi Consent Model changes
- [x] Tested in CODE with hard coded values 

## Images

**NEW:** 
image

**OLD:**
image

## Accessibility

- [ ] [Tested with screen reader](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#screen-readers)
- [ ] [Navigable with keyboard](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#keyboard-navigation)
- [ ] [Colour contrast passed](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#use-of-colour)

### CODE REVIEW

**Comment by vlbee** [comment]
Once #799 goes in, the Toggle component should be updated to use the one from Source v4.

File: app/client/components/identity/EmailAndMarketing/ConsentSection.tsx [hunk 1]
```diff
@@ -30,27 +31,49 @@ const softOptInSupporterEmailConsents = (
...
+					id={id}
+					key={id}
+					title={name}
+					description={description}
+					selected={subscribed}
+					onClick={clickHandler}
+				/>
+			);
+		}
+	}
```
Comment (liamness) [inline app/client/components/identity/EmailAndMarketing/ConsentSection.tsx:68]: Would be a nice-to-have, but could define the props once and spread them. e.g.

```jsx
const props = { id, key: id /* etc */ };
return <MarketingCheckbox {...props} />;
```

File: app/client/components/identity/MarketingToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+						button {
+							align-self: flex-start;
+							cursor: pointer;
+						}
+					`}
+					label={title}
+					labelPosition="left"
+					labelId={id}
+					checked={!!selected}
+					onClick={(_) => {
```
Comment (liamness) [inline app/client/components/identity/MarketingToggle.tsx:63]: Can the underscore be omitted?

**Review by liamness — APPROVED** [review]
Looks good to me.

(I have only skimmed over the bits which you say are currently copied from source).

File: app/package.json [hunk 3]
```diff
@@ -109,7 +109,6 @@
     "core-js": "^3.19.1",
     "cypress": "8.5.0",
     "cypress-plugin-stripe-elements": "^1.0.2",
-    "dom-testing-library": "^2.3.2",
```
Comment (ob6160) [inline app/package.json:112]: just checking if these two removed libraries are all good to go as they're not mentioned in the pr desc

**Review by ob6160 — APPROVED** [review]

File: app/package.json (hunk 3 above)
Comment (vlbee) [reply app/package.json:112]: These are old package versions. The new library packages are installed already. 
https://www.npmjs.com/package/dom-testing-library
https://www.npmjs.com/package/react-testing-library

Comment (liamness) [reply app/package.json:112]: They moved everything to the `@testing-library` org on npm.

**Pull request merged**